pkg/networkpolicy/packet.go

```diff
@@ -73,7 +73,7 @@ func parsePacket(b []byte) (packet, error) {
 		t.proto = v1.ProtocolSCTP
 		dataOffset = hdrlen + 8
 	default:
-		return t, fmt.Errorf("unknown protocol %d", protocol)
+		return t, fmt.Errorf("IP family %s unknown transport protocol %d", t.family, protocol)
```

Comment:
You don't want to log an error for every ICMPv6 message. You should add a `case 58:` for that.
Probably `parsePacket` needs another return value for "not an error, but just accept the packet".

(It seems that your callback only gets called for IP and IPv6 protocol packets, right? And, in particular, _not_ ARP and ICMP packets, which are separate L3 protocols, unlike ICMPv6/NDP, which are subprotocols of IPv6.)

Reply:
see the next commit, only TCP , UDP and SCTP packets are sent to the userspace

Reply:
At the moment I think all fragments will also have an unknown protocol. But I am about to fix that (#15)

Reply:
> see the next commit, only TCP , UDP and SCTP packets are sent to the userspace

That may fix the fragment problem. The kernel will reassemble packets before sending them to user-space (perhaps it already does in which case #15 isn't a kind/bug after all). Ref